### ISSUE TITLE
[receiver/mongodb] Flaky test integration test

### ISSUE THREAD

**Issue by atoulme** [open-telemetry/opentelemetry-collector-contrib#16273]
## Description
See this build https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3449973778/jobs/5758295362

## Steps to Reproduce
Run the build
## Expected Result
The build passes
## Actual Result
The build fails with:
```
=== CONT  TestMongodbIntegration/Running_mongodb_4.0
2022/11/12 07:17:24 Starting container id: ca7d386d6c49 image: 95692a7a-3943-4563-86e5-d6cbf919c04d:0383459b-8a1e-4bcf-8d03-f344fa1ba84c
2022/11/12 07:17:24 Waiting for container id ca7d386d6c49 image: 95692a7a-3943-4563-86e5-d6cbf919c04d:0383459b-8a1e-4bcf-8d03-f344fa1ba84c
2022/11/12 07:17:26 Container is ready id: ca7d386d6c49 image: 95692a7a-3943-4563-86e5-d6cbf919c04d:0383459b-8a1e-4bcf-8d03-f344fa1ba84c
2022/11/12 07:17:26 Starting container id: ca7d386d6c49 image: 95692a7a-3943-4563-86e5-d6cbf919c04d:0383459b-8a1e-4bcf-8d03-f344fa1ba84c
2022/11/12 07:17:26 Waiting for container id ca7d386d6c49 image: 95692a7a-3943-4563-86e5-d6cbf919c04d:0383459b-8a1e-4bcf-8d03-f344fa1ba84c
2022/11/12 07:17:26 Container is ready id: ca7d386d6c49 image: 95692a7a-3943-4563-86e5-d6cbf919c04d:0383459b-8a1e-4bcf-8d03-f344fa1ba84c
2022/11/12 07:17:32 Starting container id: 1c1b35a3278c image: 95c1af30-27ac-4029-aeb2-e733b5f6f6ed:fe43d91d-b484-4a25-80c0-ad167bca7562
    integration_test.go:242: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/mongodbreceiver/integration_test.go:242
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/mongodbreceiver/integration_test.go:78
        	Error:      	Received unexpected error:
        	            	Error response from daemon: driver failed programming external connectivity on endpoint upbeat_dirac (d53dd1546a3bccbdf3107053769bf6962dd3fe7fec97f838ab955dc12c8cd6c2): Bind for 0.0.0.0:27217 failed: port is already allocated: failed to start container
        	Test:       	TestMongodbIntegration/Running_mongodb_4.0
--- FAIL: TestMongodbIntegration (0.00s)
    --- PASS: TestMongodbIntegration/Running_mongodb_5.0 (78.61s)
    --- PASS: TestMongodbIntegration/Running_mongodb_4.4_as_LPU (85.74s)
    --- FAIL: TestMongodbIntegration/Running_mongodb_4.0 (9.96s)
    --- PASS: TestMongodbIntegration/Running_mongodb_4.2 (72.61s)
FAIL
coverage: 74.2% of statements
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver	151.260s
```

### Collector version

latest

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

**Comment by dmitryax** [comment]
Another case https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4908159661/jobs/8764714635?pr=21534

**Comment by TylerHelmuth** [comment]
Another case: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4996825525/jobs/8950535314

**Comment by djaglowski** [comment]
Closed by #22022
